test/integration/test_main.py

```diff
@@ -134,48 +131,41 @@ def test_module_run_clean(tmp_path):
     assert rc == 0
 
 
-def test_role_run(skipif_pre_ansible28, clear_integration_artifacts):
+def test_role_run(skipif_pre_ansible28, tmp_path):
+    artifact_dir = tmp_path / "otherartifacts"
     rc = main(['run', '-r', 'benthomasson.hello_role',
                '--hosts', 'localhost',
                '--roles-path', 'test/integration/roles',
```

Comment:
FYI, the `--roles-path` argument here is doing nothing. It's not the right path, because the role is in `test/integration/project/roles`. It's getting the role because it's in the `roles` directory relative to the project (where the playbook runs), so it doesn't need to be set by the Ansible config.

I think my number 1 preference would be do make a tmp dir and `shutil.copytree` the project folder to that.

Reply:
Yep, I discovered that as well. It's because the temporary playbook is created in the project directory, and the adjacent roles are found there. I'll probably fix that in a different PR.